src/commands/debug/message.ts

```diff
@@ -18,6 +29,45 @@ export default class MessageDebugCommand extends SlashCommand {
     const origin = 'guild_id' in ctx.data ? ctx.data.guild_id : '@me';
     const target_url = `https://discord.com/channels/${origin}/${ctx.data.channel_id}/${target_id}`;
 
+    this.tryDeferredAdjustment(ctx);
+
     return ChatDebugCommand.resolveFinalPayload(rawPayload, 'message', target_url);
   }
+
+  private async tryDeferredAdjustment(ctx: CommandContext): Promise<void> {
+    if (ctx.targetMessage.author.id !== this.creator.options.applicationID) return;
```

Comment:
Move message author check to `run()` - intention to prevent unnecessary promise handling.